stable_baselines3/common/recurrent/buffers.py

```diff
@@ -115,62 +161,109 @@ def __init__(
         buffer_size: int,
         observation_space: spaces.Space,
         action_space: spaces.Space,
-        hidden_state_shape: Tuple[int, int, int, int],
+        hidden_state_example: TensorTree,
         device: Union[th.device, str] = "auto",
         gae_lambda: float = 1,
         gamma: float = 0.99,
         n_envs: int = 1,
     ):
-        self.hidden_state_shape = hidden_state_shape
-        self.seq_start_indices, self.seq_end_indices = None, None
-        super().__init__(buffer_size, observation_space, action_space, device, gae_lambda, gamma, n_envs)
+        super(RolloutBuffer, self).__init__(buffer_size, observation_space, action_space, device=device, n_envs=n_envs)
+        self.hidden_state_example = hidden_state_example
+        self.gae_lambda = gae_lambda
+        self.gamma = gamma
+
+        batch_shape = (self.buffer_size, self.n_envs)
+        device = self.device
```

Comment:
I'm confused that this is `device = self.device` instead of `self.device = device`. Can you help me understand why we are overriding the parameter here?